toolkit/pallets/block-production-log/src/lib.rs

```diff
@@ -43,26 +43,34 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type LatestBlock<T: Config> = StorageValue<_, BlockNumberFor<T>, OptionQuery>;
 
+	/// This flag keeps track of whether the pallet has been initialized by receiving non-empty inherent data.
+	/// In initialized state, the inherent is requires for every block.
+	#[pallet::storage]
+	pub type Initialized<T: Config> = StorageValue<_, bool, ValueQuery>;
+
```

Comment:
you can maybe try to use other variable to check if it was already initialized, maintaining some booleans to represents if some other state is set or not may cause errors.
You can take a look on `block-production-log`, where instead of `ValueQuery` an `OptionQuery` is utilized